cgt_calc/parsers/sharesight.py

```diff
@@ -0,0 +1,284 @@
+"""Sharesight parser."""
+from __future__ import annotations
+
+import csv
+from datetime import date, datetime
+from decimal import Decimal, InvalidOperation
+from pathlib import Path
+from typing import Final, Iterable, Iterator, List
+
+from cgt_calc.exceptions import InvalidTransactionError, ParsingError
+from cgt_calc.model import ActionType, BrokerTransaction
+
+STOCK_ACTIVITY_COMMENT_MARKER: Final[str] = "Stock Activity"
+
+
+def parse_date(val: str) -> date:
+    """Parse a Sharesight report date."""
+
+    return datetime.strptime(val, "%d/%m/%Y").date()
+
+
+def parse_decimal(val: str) -> Decimal:
+    """Convert value to Decimal."""
+    try:
+        return Decimal(val.replace(",", ""))
+    except InvalidOperation:
+        raise ValueError(f"Bad decimal: {val}") from None
+
+
+def maybe_decimal(val: str) -> Decimal | None:
+    """Convert value to Decimal."""
+    return parse_decimal(val) if val else None
+
+
+class SharesightTransaction(BrokerTransaction):
+    """Sharesight transaction.
+
+    Just a marker type for now
+    """
+
+
+class RowIterator(Iterator[List[str]]):
+    """Iterator for CSV rows that keeps track of line number."""
+
+    def __init__(self, rows: Iterable[list[str]]) -> None:  # noqa: D107
```

Comment:
IMO, trivial docstring like `Initialise RowIterator` is better than `noqa` comment.

Reply:
fixed